src/dialog-modal.module.ts

```diff
@@ -0,0 +1,41 @@
+import { CommonModule } from '@angular/common';
+import { ApplicationRef, ComponentFactoryResolver, Injector, ModuleWithProviders, NgModule } from '@angular/core';
+
+import { DialogHolderComponent } from './dialog-holder.component';
+import { DialogWrapperComponent } from './dialog-wrapper.component';
+import { DialogService, DialogServiceConfig } from './dialog.service';
+import { dialogServiceFactory } from './dialog-service.factory';
+
+@NgModule({
+    declarations: [
+        DialogHolderComponent,
+        DialogWrapperComponent
+    ],
+    providers: [
+        DialogService
+    ],
+    imports: [
+        CommonModule
+    ],
+    entryComponents: [
+        DialogHolderComponent,
+        DialogWrapperComponent
+    ]
+})
+export class DialogModalModule {
```

Comment:
Since the rest of the code is using a single name, ex.: `DialogService`, `DialogComponent`, etc.. Isn't it better to just use the name `DialogModule`?

Reply:
So i was thinking on this last night... and while we're mostly dialog right now... i was thinking to myself... would i ever search NPM for a 'dialogue' component... when really i'd search for a 'modal' one...  so i flip it back over and say should be rename to be a ModalModule, ModalService, ModalHolder, ModalWrapper etc?

Reply:
further to the point, when you consider it, 95% of the dialgue behaviour e.g.. alert, confirm etc is done by the implementer not by the component.. this is supplementing a modal system which you can place dialogues within OR you can just as easily implement a gallery.

Reply:
I agree with you, this can be anything we put into the component. Just keeping the consistency with the naming of the files is enough.

About choosing wich name convention to use, I was reading more on this topic and the primordial difference is:

- Modal: User can't do anything on the page until confirm/close, or other required interaction with the modal window;
- Dialog: Let users change focus on the page;

Reply:
I'd say then based on those definitions, or at least i guess how i'm using this component, this is more of an ngx-simple-modal (looked for a free name). My modals tend to contain forms - login, register etc. Funny thing - someone else branched not long after this one - https://www.npmjs.com/package/ngx-bootstrap-modal-ass

Reply:
angularx-modal also free

Reply:
I like `ngx-simple-modal`, also very intuitive. The way I'm using is also for login an simple forms.